File: backend/app/schemas/audit.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class AuditLogCreate(BaseModel):
    action: str
    actor: str
    target: str | None = None
    outcome: str = "success"
    summary: str | None = None
    metadata: dict[str, Any] = Field(default_factory=dict)


class AuditLogRead(AuditLogCreate):
    id: int
    created_at: datetime

File: shared/contracts/audit.ts
export type AuditLogCreate = {
  action: string;
  actor: string;
  target?: string | null;
  outcome?: string;
  summary?: string | null;
  metadata?: Record<string, unknown>;
};

export type AuditLogRead = AuditLogCreate & {
  id: number;
  created_at: string;
};
